src/main/java/com/mraof/minestuck/item/StructureScannerItem.java

```diff
@@ -1,85 +1,183 @@
+	}
+	
+	public void checkLocation(ItemStack pStack, BlockPos pos)
+	{
+		if(pos == null)
+		{
+			pStack.getTag().remove("TargetLocation");
 		} else
 		{
-			foundItem = true;
+			pStack.getTag().put("TargetLocation", NbtUtils.writeBlockPos(pos));
 		}
-		
-		if(levelIn instanceof ServerLevel level)
+	}
+	
+	/**
+	 * Check that the fuel item is set; if not, don't reduce charge.
+	 * Scanner charge is represented by durability and damage. Damage is dealt every 20 ticks.
+	 * If it runs out of charge, power it off.
+	 *
+	 * @param pStack current item
+	 * @param pEntity player entity
+	 * @param pLevel player's current level
+	 * @return Boolean check for if deactivated, used to reactivate if players still has fuel.
+	 */
+	public boolean reduceCharge(ItemStack pStack, Entity pEntity, Level pLevel)
+	{
+		if(fuelItem != null && pEntity.tickCount % 20 == 0)
 		{
-			if(foundItem || playerIn.isCreative())
+			pStack.hurt(1, pLevel.random, pEntity instanceof ServerPlayer ? (ServerPlayer) pEntity : null);
+			
+			if(!isCharged(pStack))
 			{
-				BlockPos structureBlockPos = level.findNearestMapStructure(structure, playerIn.blockPosition(), 100, false);
+				pStack.getTag().putBoolean("Powered", false);
 				
-				if(structureBlockPos != null)
-				{
-					BlockPos playerBlockPos = playerIn.blockPosition();
-					playerIn.sendSystemMessage(Component.translatable(getDescriptionId() + ".successMessage",
-							(int) Math.sqrt(playerBlockPos.distSqr(structureBlockPos.above(64)))).withStyle(ChatFormatting.AQUA));
-				} else
-				{
-					playerIn.sendSystemMessage(Component.translatable(getDescriptionId() + ".failMessage").withStyle(ChatFormatting.RED));
-				}
-			} else
+				MutableComponent message = Component.translatable("message.temple_scanner.off");
+				pEntity.sendSystemMessage(message.withStyle(ChatFormatting.DARK_GREEN));
+				
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	public boolean checkFuelNeeded(Player pPlayer, Level pLevel)
+	{
+		if(fuelItem != null)
+		{
+			ItemStack invItem = hasFuel(pPlayer);
+			
+			if (invItem != null)
+			{
+				useFuelItem(invItem, pPlayer, pLevel);
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	public ItemStack hasFuel(Player pPlayer){
+		ItemStack fuelStack = new ItemStack(fuelItem.get());
+		
+		for (ItemStack invItem : pPlayer.getInventory().items)
+		{
+			if (ItemStack.isSameItem(invItem, fuelStack))
 			{
-				playerIn.sendSystemMessage(Component.translatable(getDescriptionId() + ".noFuelMessage").withStyle(ChatFormatting.RED));
+				return invItem;
 			}
 		}
-		return InteractionResultHolder.success(item);
+		return null;
+	}
+	
+	public void useFuelItem(ItemStack invItem, Player pPlayer, Level pLevel)
+	{
+		if(!pLevel.isClientSide)
+		{
+			invItem.shrink(1);
+		}
+		pLevel.playSound(pPlayer, pPlayer.getX(), pPlayer.getY(), pPlayer.getZ(), SoundEvents.BLAZE_SHOOT, SoundSource.AMBIENT, 0.4F, 2F);
 	}
```

Comment:
does not trigger when in survival, meaning the item cannot be refueled once used